cli/npm/dsl.d.ts

```diff
@@ -72,7 +73,10 @@ interface Grammar<
    * a *descending* ordering. Names listed earlier in one of these arrays
    * have higher precedence than any names listed later in the same array.
    */
-  precedences?: () => String[][],
+  precedences?: (
+    $: GrammarSymbols<RuleName | BaseGrammarRuleName>,
+    previous: RuleOrLiteral[][],
+  ) => RuleOrLiteral[][],
```

Comment:
Why you decided that there possible to use references? This is not stated in the description too.